### PR TITLE
Update interfaces and structs for admin use

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Update interfaces and structs for admin use

**Why?**
Expose those interfaces and structs for future admin use

**How did you test it?**
No logic changed

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: common/persistence/serialization/serializer.go [hunk 1]
```diff
@@ -316,151 +316,151 @@ func (t *serializerImpl) ShardInfoFromBlob(data *commonpb.DataBlob, clusterName
...
 
-func (t *serializerImpl) ReplicationTaskFromBlob(data *commonpb.DataBlob) (*replicationspb.ReplicationTask, error) {
+func (t *SerializerImpl) ReplicationTaskFromBlob(data *commonpb.DataBlob) (*replicationspb.ReplicationTask, error) {
 	result := &replicationspb.ReplicationTask{}
-	return result, proto3DecodeBlob(data, result)
+	return result, Proto3DecodeBlob(data, result)
 }
 
-func proto3DecodeBlob(data *commonpb.DataBlob, result proto.Message) error {
+func Proto3DecodeBlob(data *commonpb.DataBlob, result proto.Message) error {
```
Comment (yiminc) [inline common/persistence/serialization/serializer.go:463]: remove 3 from the method name.

File: common/persistence/serialization/serializer.go [hunk 2]
```diff
@@ -515,13 +515,13 @@ func encodeBlob(o proto.Message, encoding enumspb.EncodingType) (*commonpb.DataB
...
 	case enumspb.ENCODING_TYPE_PROTO3:
-		return proto3EncodeBlob(o, enumspb.ENCODING_TYPE_PROTO3)
+		return Proto3EncodeBlob(o, enumspb.ENCODING_TYPE_PROTO3)
 	default:
 		return nil, NewUnknownEncodingTypeError(encoding)
 	}
 }
 
-func proto3EncodeBlob(m proto.Message, encoding enumspb.EncodingType) (*commonpb.DataBlob, error) {
+func Proto3EncodeBlob(m proto.Message, encoding enumspb.EncodingType) (*commonpb.DataBlob, error) {
```
Comment (yiminc) [inline common/persistence/serialization/serializer.go:524]: remove 3 from method name

**Review by yiminc — APPROVED** [review]

File: common/persistence/serialization/serializer.go [hunk 3]
```diff
@@ -121,21 +121,21 @@ type (
 		encodingType enumspb.EncodingType
 	}
 
-	serializerImpl struct {
+	SerializerImpl struct {
```
Comment (yycptt) [inline common/persistence/serialization/serializer.go:124]: `NewSerializer` is already exported. What's the use case for exporting the underlying implementation?

File: service/history/timerQueueAckMgr.go [hunk 4]
```diff
@@ -118,17 +118,17 @@ func (t timerKeys) Swap(i, j int) {
 
 // Less implements sort.Interface
 func (t timerKeys) Less(i, j int) bool {
-	return compareTimerIDLess(&t[i], &t[j])
+	return CompareTimerIDLess(&t[i], &t[j])
 }
 
-func newTimerKey(time time.Time, taskID int64) *timerKey {
+func NewTimerKey(time time.Time, taskID int64) *timerKey {
```
Comment (yycptt) [inline service/history/timerQueueAckMgr.go:124]: Shall we deprecate the TimerKey and use `tasks.Key` instead?

Comment (yux0) [reply service/history/timerQueueAckMgr.go:124]: Yes. Good to know. I will update this.

**Pull request merged**